tiled/client/cache.py

```diff
@@ -74,20 +89,55 @@ class Reservation:
     using `ensure_released()`.
     """
 
-    def __init__(self, url, etag, lock, load_content):
+    def __init__(self, url, headers, renew, lock, load_content):
         self.url = url
-        self.etag = etag
+        self.headers = headers
+        self._renew = renew
         self._lock = lock
         self._lock_held = True
         self._load_content = load_content
         lock.acquire()
 
+    @property
+    def etag(self):
+        return self.headers["ETag"]
+
+    @property
+    def expires(self):
+        expires = self.headers.get("Expires")
+        if expires is not None:
+            return datetime.strptime(expires, HTTP_EXPIRES_HEADER_FORMAT)
```

Comment:
```suggestion
            return datetime.strptime(expires, HTTP_EXPIRES_HEADER_FORMAT)
        return None
```

I think the explicit return is good for readability here.

Reply:
Done